lib/payrix/resource/mixin/attr_casters.rb

```diff
@@ -0,0 +1,43 @@
+module Payrix
+  module Resource
+    module Mixin
+      module AttrCasters
+        def self.included(klass)
+          klass.extend(ClassMethods)
+        end
+
+        module ClassMethods
+          # Use this mixin to dynamically define a getter that casts the specified attribute to an integer. Note that
+          # methods are defined (and overridden) in the order they appear, so make sure to use this *after* any
+          # additional usage of methods like `.attr_reader` or `.attr_accessor`.
+          #
+          # In addition to defining a integer-casting getter wrapper around the attribute, it also creates another
+          # method, `#{attribute}_raw_value`, which gives the pre-cast value.
+          #
+          #   class Txns < Base
+          #     include Mixin::AttrCasters
+          #
+          #     def initialize(params)
+          #       super(params, [:status, :date, ...])
+          #
+          #       @resource_name = 'txns'
+          #     end
+          #
+          #     attr_accessor :status, :date, ...
+          #     attr_cast_to_integer :status
+          #   end
+          #
+          #   transaction = Txns.new(status: '1')
+          #   transaction.status # => 1
+          #   transaction.status_raw_value # => "1"
+          def attr_cast_to_integer(*attrs)
+            attrs.each do |attr|
+              define_method(attr) { self.instance_variable_get("@#{attr}").to_i }
+              define_method("#{attr}_raw_value") { self.instance_variable_get("@#{attr}") }
+            end
+          end
```

Comment:
There isn't too much going on here, we are defining 2 methods dynamically.

- The first one on line 35 is `#{attribute}`, which will redefine the reader defined by `.attr_accessor`, and converts the instance variable to an integer with `#to_i`.
- The second one on line 36 is `#{attribute}_raw_value`, which provides access (if needed) to the original value for posterity.